src/main/kotlin/no/nav/syfo/dialogmote/tilgang/DialogmoteTilgangService.kt

```diff
@@ -64,11 +66,19 @@ class DialogmoteTilgangService(
         token: String,
         callId: String,
     ): Boolean {
-
         return if (hasAccessToDialogmotePerson(personIdentNumber, token, callId)) {
             val kontaktinfo = kontaktinformasjonClient.kontaktinformasjon(personIdentNumber, token, callId)
-            val isDigitalVarselAllowed = kontaktinfo?.kontaktinfo?.get(personIdentNumber.value)?.kanVarsles ?: false
-            isDigitalVarselAllowed
+            val isDigitalVarselAllowed = kontaktinfo?.kontaktinfo?.isDigitalVarselAllowed(personIdentNumber)
+            return if (isDigitalVarselAllowed == false) {
```

Comment:
Er ikke "return" her redundant?

Reply:
Ja, synes det er litt forvirrende at hasAccessToDialogmoteInnkalling gjør mer enn å sjekke tilgang. Føler at sjekk på om person kan varsles burde separares fra tilgangssjekk på en eller annen måte. Men kanskje ikke så lett siden man må ha tilgang for å få sjekket kontaktinfo?

Virker også litt rart å få både "Denied Veileder access to create NewDialogmoteTidSted for moteUUID" og ""Denied access create or update DialogmoteInnkalling: DigitalVarsel is not allowed" f.eks. når man forsøker opprette innkalling på arbeidstaker som ikke kan varsles?

Reply:
Jo, har fikset det nå 👍

Reply:
@andersrognstad Ser den.
1. Må ha tilgang til person for å sjekke kontaktinfo ja. Kan jo alltids lage en ny metode som beskriver det eksplisitt, men det trengs en tilgangsjekk som sjekker både det ene og det andre.

2.  Mtp logging så har jeg lagt til en ny commit nå, hvor det ikke logges noe i API-endepunktene, men heller et felles sted avhengig av hva som er årsaken. Hva synes du om det?

`} else {
            log.warn(
                "Denied access create or update DialogmoteInnkalling: No access to person, {}",
                callIdArgument(callId),
            )
            false
        }`

Reply:
Bedre det 👍 Kan også vurdere om man skal splitte i to metoder ala bare sjekke tilgang i hasAccessToDialogmoteInnkalling() og i Api-metodene gjøre en dialogmoteTilgangService.kanVarsles() e.l. hvis resultatet av hasAccess er true f.eks. Evt bare rename hasAccessToDialogmoteInnkalling for å beskrive at den sjekker flere ting...

Reply:
Endret litt navnet på funksjonen nå.